### PR TITLE
New class for prompts in enquirer and Integration Tests.

### PR DESCRIPTION
**Pull request by 1AmNegan**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

- Add a new class for Prompt in enquirer.
- Added Integration tests for init command.
- Sandboxed sinon spies.

### CODE REVIEW

**Comment by prasadtalasila** [comment]
@1AmNegan please rebase your autolabcli-ng branch to the the latest commits on the autolabcli-ng branch. It is also better to have one single commit for the pull request unless there is a strong reason for multiple commits.

File: lib/cli/input/init.js [hunk 1]
```diff
@@ -1,32 +1,26 @@
...
-  message: 'Enter your password:',
-  validate: function(value) {
+const getLengthValidator = (invalidMessage) => {
+  return (value) => {
     if (value.length) {
       return true;
     }
     else {
-      return 'Please enter your password';
+      return invalidMessage;
```
Comment (prasadtalasila) [inline lib/cli/input/init.js]: This return message does not seem consistent with the previous approach of giving the prompt again. Am I missing something in the code?

File: lib/utils/Prompt.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Prompt {
+  constructor (name, type, message, validate) {
+    this.name = name;
+    this.type = type;
+    this.message = message;
+    this.validate = validate;
+  }
```
Comment (prasadtalasila) [inline lib/utils/Prompt.js]: we discussed about another method to perform dynamic addition of <key,value> to Prompt object. The method can have the following signature.

```
add (key, value)
```

File: test/lib/controller/init.js [hunk 3]
```diff
@@ -16,9 +16,14 @@ chai.should();
 describe('For init controller', () => {
```
Comment (prasadtalasila) [inline test/lib/controller/init.js]: I don't understand the rest of the code in this module. We will have to meet and discuss the same.

File: lib/cli/input/init.js (hunk 1 above)
Comment (1AmNegan) [reply lib/cli/input/init.js]: The validate function is the same for both the prompts except for the message displayed when invalid. Hence I'm passing this message as a parameter.

Comment (prasadtalasila) [reply lib/cli/input/init.js]: @1AmNegan Good use of closures. Nice code.

**Comment by prasadtalasila** [comment]
@1AmNegan Please correct the format and verification of all the test doubles.

**Pull request merged**